src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,121 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_LIST = {};
+    private Object[] elementArray;
+    private int size;
+
+    public ArrayList() {
+        this.elementArray = EMPTY_LIST;
+    }
+
+    private Object[] grow(int minCapacity) {
+        int oldCapacity = elementArray.length;
+        int newCapacity = minCapacity + (oldCapacity >> 1);
+        if (oldCapacity > 0) {
+            return elementArray = Arrays.copyOf(elementArray, newCapacity);
+        }
+        return elementArray = new Object[Math.max(DEFAULT_CAPACITY, newCapacity)];
+    }
+
+    private Object[] grow() {
```

Comment:
it looks like it's redundant to have 2 grow() methods, you never pass custom values there